src/bson_value.ts

```diff
@@ -10,8 +11,16 @@ export abstract class BSONValue {
     return BSON_MAJOR_VERSION;
   }
 
+  [Symbol.for('nodejs.util.inspect.custom')](
+    depth?: number,
+    options?: unknown,
+    inspect?: InspectParameterFn
+  ): string {
+    return this.inspect(depth, options, inspect);
+  }
+
   /** @public */
-  public abstract inspect(): string;
+  public abstract inspect(depth?: number, options?: unknown, inspect?: InspectParameterFn): string;
```

Comment:
Would it be possible to provide sensible defaults here instead of making all the arguments optional?

```suggestion
  public abstract inspect(depth: number = 3, options: unknown = { stylize: basicInspectParameterFn }, inspect: InspectParameterFn = basicInspectParameterFn): string;
```

Then all the implementations in each BSON type don't need to worry about setting defaults

Reply:
It would be nice if we were able to make some sensible defaults here! However, to my knowledge, since this is an abstract method we can't assign values to arguments without the `"A parameter initializer is only allowed in a function or constructor implementation."` error popping up.